omod/src/main/java/org/bahmni/module/bahmniOfflineSync/strategy/IDBasedSyncStrategy.java

```diff
@@ -184,4 +204,36 @@ private boolean isOfflineConceptEvent(String eventUuid) {
         final Concept offlineConcept = conceptService.getConceptByName("Offline Concepts");
         return offlineConcept != null && offlineConcept.getSetMembers().contains(concept);
     }
+
+    protected String getFilterForLocalIdentifierSource(String identifier) {
+        Session session = null;
+        try {
+            session = sessionFactory.getCurrentSession();
+            session.beginTransaction();
+            String prefix = session.doReturningWork(new ReturningWork<String>() {
+                @Override
+                public String execute(Connection conn) throws SQLException {
+                    PreparedStatement pStmt = null;
+                    try {
+                        pStmt = conn.prepareStatement(identifierPrefixQuery);
+                        pStmt.setString(1, identifier);
+                        ResultSet resultSet = pStmt.executeQuery();
+                        if(resultSet.next())
```

Comment:
I think here we can use ternary condition

Reply:
Updated...